File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/managers/analytics/AnalyticsModule.java
package com.androidcollider.easyfin.common.managers.analytics;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ihor Bilous
 */

@Module
public class AnalyticsModule {

    @Provides
    @Singleton
    AnalyticsManager provideAnalyticsManager(Context context) {
        return new AnalyticsManager(context);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/managers/analytics/AnalyticsModule.kt
package com.androidcollider.easyfin.common.managers.analytics

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author Ihor Bilous
 */
@Module
class AnalyticsModule {
    @Provides
    @Singleton
    fun provideAnalyticsManager(context: Context): AnalyticsManager {
        return AnalyticsManager(context)
    }
}